### PR TITLE
fix Issue 24331 - @nogc and GC.disable() are often confused

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

File: spec/function.dd [hunk 1]
```diff
@@ -3752,6 +3752,14 @@ $(H2 $(LNAME2 nogc-functions, No-GC Functions))
             }
             ---
 
+	$(BEST_PRACTICE Since a function marked `@nogc` will not do any GC allocations,
+	that implies it will not cause any GC collections to run. However,
+	another thread may still allocate with the GC and trigger a collection.
+	The recommended way to prevent GC collections from being run is to call
+	$(LINK2 https://dlang.org/phobos/core_memory.html#GC.disable, core.memory.GC.disable())
+	instead. This will stop collections from being run in any thread until a corresponding
+	call to `core.memory.GC.enable()` is run. GC allocations can also still be performed
```
Comment (rikkimax) [inline spec/function.dd:3761]: > GC allocation can still be performed while `GC.disable()` is in effect.

Original sentence isn't reading right.

Comment (WalterBright) [reply spec/function.dd:3761]: I don't know what you mean?

Comment (WalterBright) [reply spec/function.dd]: Oh I see now

**Review by tgehr — APPROVED** [review]
Looks good to me. I think Adam Wilson and deadalnix are in the mid term also trying to roll back the guarantee that there are no background collections though.

**Comment by WalterBright** [comment]
Well, when/if the GC changes the documentation will have to change, too!

**Pull request merged**